src/Network/HTTP/StackClient.hs

```diff
@@ -83,3 +96,52 @@ withResponse
   :: (MonadUnliftIO m, MonadIO n)
   => Request -> (Response (ConduitM i Strict.ByteString n ()) -> m a) -> m a
 withResponse = Network.HTTP.Simple.withResponse . setUserAgent
+
+-- | Set the user-agent request header
+setGithubHeaders :: Request -> Request
+setGithubHeaders = setRequestHeader "Accept" ["application/vnd.github.v3+json"]
+
+-- | Download the given URL to the given location. If the file already exists,
+-- no download is performed. Otherwise, creates the parent directory, downloads
+-- to a temporary file, and on file download completion moves to the
+-- appropriate destination.
+--
+-- Throws an exception if things go wrong
+download :: HasTerm env
+         => Request
+         -> Path Abs File -- ^ destination
+         -> RIO env Bool -- ^ Was a downloaded performed (True) or did the file already exist (False)?
+download req dest = Download.download (setUserAgent req) dest
```

Comment:
I'm not at all convinced this is a good idea, but: since the only purpose of this wrapper seems to be to set the user agent, perhaps we should have a `HasHttp`-style typeclass in `http-download` which includes a user agent setting instead. Then we know we won't ever accidentally forget to set it, and we can drop this wrapper module entirely.

Reply:
I like the idea but at the same time it seems to add noticeable amount of code and it will resolve not very critical issue so I'd skip it in this PR